linux/mcompile/keymap/mc_kmxfile.cpp

```diff
@@ -1,297 +1,512 @@
-
 #include "mc_kmxfile.h"
+#include <typeinfo>
+
+#define CERR_None                                          0x00000000
+#define CERR_CannotAllocateMemory                          0x00008004
+#define CERR_UnableToWriteFully                            0x00008007
+#define CERR_SomewhereIGotItWrong                          0x00008009
+
+KMX_BOOL KMX_VerifyKeyboard(LPKMX_BYTE filebase, KMX_DWORD sz);
+
+LPKMX_KEYBOARD KMX_FixupKeyboard(PKMX_BYTE bufp, PKMX_BYTE base, KMX_DWORD dwFileSize);
+
+KMX_BOOL KMX_SaveKeyboard(LPKMX_KEYBOARD kbd, PKMX_WCHAR filename) {
```

Comment:
We can use `char*` (or `KMX_CHAR*`) as data type for `filename`.

Reply:
OK done: see PR https://github.com/keymanapp/keyman/pull/11879